apis/vmware/v1beta1/vspheremachine_types.go

```diff
@@ -34,7 +35,95 @@ type VSphereMachineVolume struct {
 	StorageClass string `json:"storageClass,omitempty"`
 }
 
+// RouteSpec defines a static route for a guest.
```

Comment:
nit: please order the types topologically. Just easier to read view, i.e.
* VSphereMachineSpec
* VSphereMachineNetworkSpec
* InterfacesSpec
* InterfaceSpec
* InterfaceProperty
* PartialObjectRef
* RouteSpec

(let's only do this here, I think in the other file it makes sense to keep the order the same as in vm-operator to make diffing easier)